common/lib/dependabot/file_parsers/base.rb

```diff
@@ -49,7 +49,7 @@ def initialize(dependency_files:, source:, repo_contents_path: nil,
         check_required_files
       end
 
-      sig { abstract.returns(Dependabot::DependencyFile) }
+      sig { abstract.returns(T::Array[Dependabot::Dependency]) }
```

Comment:
Looks like it might actually be:

```suggestion
      sig { abstract.returns(Dependabot::DependencySet) }
```

Reply:
The ones I looked at all used `DependencySet`, but actually were returning `Array[Dependabot::Dependency]` because they called `dependency_set.dependencies` (see [elm](https://github.com/dependabot/dependabot-core/blob/main/elm/lib/dependabot/elm/file_parser.rb#L22) or [python](https://github.com/dependabot/dependabot-core/blob/81ac70fde4c400110310f1b3804b09edfe76bcec/python/lib/dependabot/python/file_parser.rb#L48)).  If there's an ecosystem(s) that returns DependencySet you know of maybe we need to do `T.any(DependencySet, Array[Dependency])`?

Reply:
We talked offline and it looks like they're all returning Array[Dependency].